stanford/roles/cluster/tasks/instance.yml

```diff
@@ -25,38 +25,31 @@
 - debug:
     var: item
   with_items: "{{ ec2.instances | default([])}}"
-- name: Add new instance to host group
```

Comment:
Since we now reference the IP address in multiple places below,
we move the conditional logic up here to decide which to use (private or public) and just set that in a variable.
The variable can then be used unconditionally .